room/runtime/src/test/java/androidx/room/SQLiteCopyOpenHelperTest.kt

```diff
@@ -180,6 +184,30 @@ class SQLiteCopyOpenHelperTest {
         assertEquals(1, exceptions.size)
     }
 
+    @Test
+    fun gzippedDatabaseProperlyUnzips() {
+        val copyFile = tempDirectory.newFile("toCopy.db")
+        writeDatabaseVersion(copyFile)
+        val gzippedCopyFile = tempDirectory.newFile("toCopy.db.gz")
+        gzipFile(copyFile, gzippedCopyFile)
+        setupMocks(tempDirectory.root, copyFile)
+
+        val openHelper = SQLiteCopyOpenHelper(
+            context,
+            null,
+            gzippedCopyFile,
+            true,
+            DB_VERSION,
+            delegate
+        ).apply { setDatabaseConfiguration(configuration) }
+        openHelper.writableDatabase
+        openHelper.writableDatabase
+
+        verify(delegate).databaseName
+        verify(context).getDatabasePath(DB_NAME)
+        verifyNoMoreInteractions(configuration)
```

Comment:
i am sad about this - i copied the single `singleCopy` test, but lost the part that makes sure the file is only read once - would love some suggestions on how to make this test better.

Reply:
These host-side unit tests were really about the file & thread locking mechanism, I say a better test would be one in the integration app, here: https://github.com/androidx/androidx/blob/androidx-master-dev/room/integration-tests/testapp/src/androidTest/java/androidx/room/integration/testapp/test/PrepackageTest.java

You can simply copy one of the existing databases and just gzip it: https://github.com/androidx/androidx/tree/androidx-master-dev/room/integration-tests/testapp/src/main/assets/databases

Reply:
done, thanks!